coderd/workspaceagents.go

```diff
@@ -396,11 +397,9 @@ func (api *API) workspaceAgentLogs(rw http.ResponseWriter, r *http.Request) {
 	}
 	go httpapi.Heartbeat(ctx, conn)
 
-	ctx, wsNetConn := codersdk.WebsocketNetConn(ctx, conn, websocket.MessageText)
-	defer wsNetConn.Close() // Also closes conn.
+	encoder := wsjson.NewEncoder[[]codersdk.WorkspaceAgentLog](conn, websocket.MessageText)
+	defer encoder.Close(websocket.StatusGoingAway)
```

Comment:
Why do we use going away instead of normal closure? It's kind of an error state and a divergence from before (i.e. status from calling `wsNetConn.Close()`).

Reply:
I chose GoingAway because one likely reason for the server closing the connection is that it's shutting down.  We could also be closing because there is a new build and this agent is no longer current.

If we actually used status codes for anything we'd want to send different codes in these cases, but we don't.  I can change it to match the old `wsNetConn` for consistency.